### PR TITLE
Remove block/entity explosion damage | Remove gamemode limits

### PR DESCRIPTION
**Pull request by ExoPlant**

### CODE REVIEW

**Comment by ExoPlant** [comment]
The PR/commit names speak for themselves

**Comment by SekoiaTree** [comment]
I haven't tested this for myself but... wouldn't cancelling setAbilities make it so you're locked to whatever abilities you spawn with?

**Comment by ExoPlant** [comment]
> I haven't tested this for myself but... wouldn't cancelling setAbilities make it so you're locked to whatever abilities you spawn with?

Oh, maybe, let me try that

**Comment by ExoPlant** [comment]
Yea, it does, welp, this is what the PR does now 😎

**Comment by SekoiaTree** [comment]
lmaoo this is even worse
First, I don't think this compiles? You're missing an argument in your abilities mixin (`   public void setAbilities(PlayerAbilities abilities) {` in yarn
Second, setAbilities is *very* broken
`
   public boolean invulnerable;
   public boolean flying;
   public boolean allowFlying;
   public boolean creativeMode;
   public boolean allowModifyWorld = true;
   private float flySpeed = 0.05F;
   private float walkSpeed = 0.1F;`
Default values of PlayerAbilities. This means that no player can fly, no player can instabreak, no player is invulnerable, and all players (including adventure) can break blocks.
So basically, you made it so everyone is in survival... forever.

**Comment by SekoiaTree** [comment]
Also, I learned that GameMode code in mc is not very pretty. Enum values are used as gamemodes, with many magic values.

**Comment by ExoPlant** [comment]
It does compile, I have tested it

**Comment by ExoPlant** [comment]
This works, though, please merge

**Comment by anatom3000** [comment]
Ok
